### PR TITLE
Remove tocimobject to build params, extend tests,  fix issues in test code

### PR DESCRIPTION
**Pull request by KSchopmeyer**
Ready for review and merge

Added the changes we wanted to make so this is effectively a new pr.

Rebased on pr#132 to pick up change to pytest-cov version

See commit for details

### CODE REVIEW

File: pywbemcli/_common.py [hunk 1]
```diff
@@ -282,12 +282,14 @@ def create_cimvalue(cim_type, value_str, is_array):
     CIMValue or list of CIMValue elements.
 
     Parameters:
-      CIM_Type: TODO
-        CIMType for this value.
+      CIM_Type: (:term: `string`)
+        CIMType for this value. The CIM data type name for the CIM object.
+        See :ref:`CIM data types` for valid type names.
 
       Value_str (:term: `string`):
```
Comment (andy-maier) [inline pywbemcli/_common.py]: Change "Value_str" in the description to "value_str" to match the lexical case of the parameter.

File: pywbemcli/_common.py [hunk 2]
```diff
@@ -282,12 +282,14 @@ def create_cimvalue(cim_type, value_str, is_array):
...
-        String defining the input to be parsed.
-        TODO cover escapes
+        String defining the input to be parsed. This may be a quoted string
+        to provide for space characters in the string in which case the
+        quote character used must be excaped if used within the string.
```
Comment (andy-maier) [inline pywbemcli/_common.py]: This description suggests that the surrounding quotes are present in the string that is passed as a value. Is that really intended?

File: pywbemcli/_common.py [hunk 3]
```diff
@@ -449,9 +473,10 @@ def split_array_value(string, delimiter):
 
 
 def split_value_str(string, delimiter):
-    """Spit a string based on a delimiter character bypassing escaped
-       instances of the delimiter.  This is a generator function in that
-       it yields each time it separates a value.
+    """
```
Comment (andy-maier) [inline pywbemcli/_common.py]: Typo "Spit".
Indentation of the first paragraph should be one more.
Indentation of the last paragraph should now also be adjusted to match the first.

File: pywbemcli/_common.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbemcli/_common.py]: At this point that comment should have not been in the description.  It is a string.

File: pywbemcli/_common.py (hunk 3 above)
Comment (KSchopmeyer) [reply pywbemcli/_common.py]: DONE

**Comment by KSchopmeyer** [comment]
The original version of this pr was effectively empty.  We had converted one use of tocimobj in an earlier pr but had not converted the one used for invokemethod.  This makes that change and since there were no tests for invokemethod, adds an initial set of tests for invokemethod.

**Comment by KSchopmeyer** [comment]
Found the issue and it was us. pywbem 0.12.0 causes the problem.  Updating the minimum to 0.12.5 eliminated the problem

**Pull request merged**